File: viewer/lib/models/lesson.dart
class Lesson {
  final String id;
  final String title;
  final String slug;
  final String? videoUrl;
  final String? content;
  final int orderIndex;
  final String techniqueId;
  final String? techniqueName;
  final String? positionName;
  final String? techniqueVideoUrl;

  Lesson({
    required this.id,
    required this.title,
    required this.slug,
    this.videoUrl,
    this.content,
    required this.orderIndex,
    required this.techniqueId,
    this.techniqueName,
    this.positionName,
    this.techniqueVideoUrl,
  });

  factory Lesson.fromJson(Map<String, dynamic> json) {
    return Lesson(
      id: json['id'] as String,
      title: json['title'] as String,
      slug: json['slug'] as String,
      videoUrl: json['video_url'] as String?,
      content: json['content'] as String?,
      orderIndex: (json['order_index'] as num).toInt(),
      techniqueId: json['technique_id'] as String,
      techniqueName: json['technique_name'] as String?,
      positionName: json['position_name'] as String?,
      techniqueVideoUrl: json['technique_video_url'] as String?,
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'title': title,
        'slug': slug,
        'video_url': videoUrl,
        'content': content,
        'order_index': orderIndex,
        'technique_id': techniqueId,
      };
}

File: app/schemas/lesson.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class LessonRead(BaseModel):
    """Schema de leitura para Lesson."""

    id: UUID
    academy_id: UUID | None = None
    title: str
    slug: str
    video_url: str | None
    content: str | None
    order_index: int
    base_points: int | None = None
    technique_id: UUID
    created_at: datetime
    technique_name: str | None = None
    position_name: str | None = None
    technique_video_url: str | None = None

    class Config:
        from_attributes = True


class LessonCreate(BaseModel):
    """Schema para criar Lesson. Slug opcional (gerado automaticamente a partir do título)."""

    technique_id: UUID
    title: str
    slug: str | None = None
    video_url: str | None = None
    content: str | None = None
    order_index: int = 0
    base_points: int | None = None


class LessonUpdate(BaseModel):
    """Schema para atualização parcial de Lesson."""

    technique_id: UUID | None = None
    title: str | None = None
    slug: str | None = None
    video_url: str | None = None
    content: str | None = None
    order_index: int | None = None
    base_points: int | None = None